test/integration/preemption_toleration_test.go

```diff
@@ -0,0 +1,310 @@
+			priorityClass: makeTestPriorityClass(testPriorityClassName, testPriority, map[string]string{
+				preemptiontoleration.AnnotationKeyMinimumPreemptablePriority: "a",
+			}),
+			preemptor: makePod(
+				st.MakePod().Name("p").PreemptionPolicy(v1.PreemptNever).Container(pauseImage).ZeroTerminationGracePeriod().Priority(testPriority + 1),
+			).ResourceRequests(podRequest).Obj(),
+			waitPreemptorCreation: 30 * time.Second, // FIXME(everpeace): how to simulate forever??
+			canTolerate:           true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// prepare test cluster
+			registry := fwkruntime.Registry{preemptiontoleration.Name: preemptiontoleration.New}
+			cfg, err := util.NewDefaultSchedulerComponentConfig()
+			if err != nil {
+				t.Fatal(err)
+			}
+			cfg.Profiles[0].Plugins.PostFilter = schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: preemptiontoleration.Name},
+				},
+				Disabled: []schedapi.Plugin{
+					{Name: "*"},
+				},
+			}
+			cfg.Profiles[0].PluginConfig = append(cfg.Profiles[0].PluginConfig, schedapi.PluginConfig{
+				Name: preemptiontoleration.Name,
+				Args: &schedconfig.PreemptionTolerationArgs{
+					MinCandidateNodesPercentage: 10,
+					MinCandidateNodesAbsolute:   100,
+				},
+			})
+			testCtx := util.InitTestSchedulerWithOptions(
+				t,
+				testutils.InitTestAPIServer(t, "sched-preemptiontoleration", nil),
+				true,
+				scheduler.WithProfiles(cfg.Profiles[0]),
+				scheduler.WithFrameworkOutOfTreeRegistry(registry),
+				scheduler.WithPodInitialBackoffSeconds(int64(0)),
+				scheduler.WithPodMaxBackoffSeconds(int64(0)),
+			)
+			defer testutils.CleanupTest(t, testCtx)
+
+			cs, ns := testCtx.ClientSet, testCtx.NS.Name
+
+			// Create test node
+			if _, err := cs.CoreV1().Nodes().Create(testCtx.Ctx, node, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("failed to create node: %v", err)
+			}
+
+			// create test priorityclass for evaluating PreemptionToleration policy
+			if _, err := cs.SchedulingV1().PriorityClasses().Create(testCtx.Ctx, tt.priorityClass, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("failed to create priorityclass %s: %v", tt.priorityClass.Name, err)
+			}
+
+			// Create victim candidate pod and it should be scheduled
+			if _, err := cs.CoreV1().Pods(ns).Create(testCtx.Ctx, victimCandidate, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("failed to create victim candidate pod %q: %v", victimCandidate.Name, err)
+			}
+			if err := wait.Poll(1*time.Second, 60*time.Second, func() (bool, error) {
+				return podScheduled(cs, ns, victimCandidate.Name), nil
+			}); err != nil {
+				t.Fatalf("victim candidate pod %q failed to be scheduled: %v", victimCandidate.Name, err)
+			}
+			// simulate pod scheduled time
+			if err := wait.Poll(1*time.Second, 60*time.Second, func() (bool, error) {
```

Comment:
Why use Poll() instead of a simple for loop to update the pod's condition?

Reply:
This wanted to assert to update the victim's scheduled time eventually with timeout.
I think we need to handle timeout, interval when using for loop.